TheraLang.Web/Controllers/ResourceController.cs

```diff
@@ -88,14 +79,14 @@ public async Task<IActionResult> PutResource(int id, [FromBody] ResourceViewMode
                 throw new ArgumentException($"{nameof(resourceModel)} is not valid");
             }
 
-            var UserId = User.Claims.GetUserId();
-            if (UserId == null) return BadRequest();
-            var user = _userManager.GetUserById(UserId.Value);
+            var authUser = await _authenticateService.GetAuthUserAsync();
+            if (authUser == null) return BadRequest();
+            var user = _userManager.GetUserById(authUser.Id);
 
             var mapper = new MapperConfiguration(cfg => cfg.CreateMap<ResourceViewModel, ResourceDto>()).CreateMapper();
             var resourceDto = mapper.Map<ResourceViewModel, ResourceDto>(resourceModel);
 
-            await _service.AddResource(resourceDto, UserId.Value);
+            await _service.AddResource(resourceDto, user.Id);
```

Comment:
why don't you pass authUser.Id here? In this case `var user = _userManager.GetUserById(authUser.Id);` isn't necessary.